### PR TITLE
CHK-1187 GEC integration

### PR DESCRIPTION
**Pull request by Fernando-Granato**
#### List of Changes

1. Made a new api for integration with GEC:
      **- POST payment-methods/:id/fee/calculate**
2. Removed old logic for psp recovery.
3. Removed api-config json mock.

#### Motivation and Context
Was modified the previous logic for retrieving PSPs with a new api that will query GEC to retrieve bundles sorted by fee.

#### How Has This Been Tested?
The tests were carried out with pagopa-ecommerce-local.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/pagopa/ecommerce/payment/methods/controller/FeeController.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    FeeService feeService;
+
+    @Override
+    public Mono<ResponseEntity<BundleOptionDto>> calculateFees(
+                                                               Mono<PaymentOptionDto> paymentOptionDto,
+                                                               Integer maxOccurrences,
+                                                               ServerWebExchange exchange
+    ) {
+        return feeService.computeFee(paymentOptionDto, maxOccurrences).map(
+                resp -> ResponseEntity.ok(resp)
```
Comment (Fernando-Granato) [inline src/main/java/it/pagopa/ecommerce/payment/methods/controller/FeeController.java:30]: Lambda can be replaced with ResponseEntity::ok 🙂

File: src/main/java/it/pagopa/ecommerce/payment/methods/application/FeeService.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                                        )
+                                        .collect(toList())
+                        )
+
+        ).flatMap(reqBody -> afmClient.getFees(reqBody, maxOccurrences))
+                .map(bo -> {
+                    /*
+                     * bo.setBundleOptions( groupBundleByPsp(bo) .values() .stream()
+                     * .map(transfersAgg -> transfersAgg.get(0)) // Get top-1 .collect(toList()) );
+                     */
```
Comment (ciuffagianluca) [inline src/main/java/it/pagopa/ecommerce/payment/methods/application/FeeService.java:65]: remove this comment

File: api-spec/api.yaml [hunk 3]
```diff
@@ -73,75 +110,63 @@ paths:
...
-            enum: [IT, EN, FR, DE, SL]
-          description: Service language
-          required: false
+            type: integer
+      requestBody:
+        $ref: "#/components/requestBodies/PostPaymentMethodPSP"
       responses:
         '200':
-          description: New payment method successfully updated
+          description: Return list of boundle returned by GEC
```
Comment (ciuffagianluca) [inline api-spec/api.yaml:147]: typo boundle is bundle

File: api-spec/api.yaml [hunk 4]
```diff
@@ -220,62 +242,108 @@ components:
...
+        transferList:
+          type: array
+          items:
+            $ref: "#/components/schemas/TransferListItem"
+      required:
+        - paymentAmount
+        - primaryCreditorInstitution
+        - transferList
+        - touchpoint
+    BundleOption:
```
Comment (infantesimone) [inline api-spec/api.yaml:270]: https://github.com/pagopa/pagopa-infra/pull/766#discussion_r1132390900

**Review by ciuffagianluca — APPROVED** [review]
LGTM 👍

**Pull request merged**